checker/bin/hive-checker.sh

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env bash
+#
+# The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+# (the "License"). You may not use this work except in compliance with the License, which is
+# available at www.apache.org/licenses/LICENSE-2.0
+#
+# This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied, as more fully set forth in the License.
+#
+# See the NOTICE file distributed with this work for information regarding copyright ownership.
+#
+
+LAUNCHER=
+# If debugging is enabled propagate that through to sub-shells
+if [[ "$-" == *x* ]]; then
+  LAUNCHER="bash -x"
+fi
+CHECKER_BIN_PATH=$(cd "$( dirname "$( readlink "$0" || echo "$0" )" )"; pwd)
+
+USAGE="Usage: alluxio-checker.sh hive [-mode USER_MODE] [-hiveurl HIVE_URL] [-name HIVE_USER_NAME] [-password HIVE_USER_PASSWORD]
+
+Prerequisites:
+    Please start hive-server2.
+    Please set the alluxio.master.hostname in your <ALLUXIO_HOME>/conf/alluxio-site.properties.
+
+Argument:
+    -hiveurl HIVE_URL is a database url of form jdbc:subprotocol:subname.
+
+Optional arguments:
+    -mode USER_MODE is one of the following integer, by default the value is 1.
+        1   use Alluxio as one option to store Hive tables.
+        2   use Alluxio as the Hive default filesystem.
+    -user HIVE_USER_NAME is the database user on whose behalf the connection is being made, by default is the system username.
+    -password HIVE_USER_PASSWORD is the user's password, if it is an empty string, you do not need to pass in anything.
+
+-h  display this help."
+
+ALLUXIO_BIN_PATH=""
+ALLUXIO_CHECKER_JAR=""
+ALLUXIO_PATH=$(cd "${CHECKER_BIN_PATH}/../../"; pwd)
+ALLUXIO_URL=""
+HIVE_USER_MODE=""
```

Comment:
I set these variables as global variables so that they can initialize in one function and use in another function.
For example, ALLUXIO_URL is given its value in main function and use in the trigger_hive function.

Reply:
I don't think that requires setting them here. For example, the following will print "hello world"

```bash
a() {
  echo $x
}

main() {
  x="hello world"
  a
}

main
```